### ISSUE TITLE
Failed to fetch warning and Fetcher failure error

### ISSUE THREAD

**Issue by scanakci** [riscv/riscv-poky#19]
Hi riscv community,

I am planning to use riscv-poky on my computer, ubuntu OS and x86_64. Detailed system:
Linux ubuntu 4.8.0-54-generic #57~16.04.1-Ubuntu SMP  x86_64 x86_64 x86_64 GNU/Linux

I typed following four lines of command in a row in my terminal:
git clone https://github.com/riscv/riscv-poky.git  (SUCCESFULLY CLONED)
cd riscv-poky
source oe-init-build-env (DID NOT GET ANY ERROR)
bitbake core-image-riscv ( causes three warnings 2 errors):

WARNING: Failed to fetch URL http://www.zlib.net/zlib-1.2.8.tar.xz, attempting MIRRORS if available
(I checked the url, it does not exist currently.)

WARNING: Failed to fetch URL git://github.com/riscv/riscv-linux.git;branch=master;destsuffix=/home/sc/Apps/riscv-poky/build/tmp/work/riscv64-poky-linux/linux-libc-headers/4.1-r0/linux-4.1.17, attempting MIRRORS if available
(No idea how to fix this warning)

ERROR: Fetcher failure: Unable to find revision 174f39501397a32c3d9c3220e3b55ec20c16303f in branch master even from upstream
(Not sure which one is the correct in branch master)

ERROR: Function failed: Fetcher failure for URL: 'git://github.com/riscv/riscv-linux.git;branch=master;destsuffix=/home/sc/Apps/riscv-poky/build/tmp/work/riscv64-poky-linux/linux-libc-headers/4.1-r0/linux-4.1.17'. Unable to fetch URL from any source.
(This url is also problematic.)

ERROR: Logfile of failure stored in: /home/sc/Apps/riscv-poky/build/tmp/work/riscv64-poky-linux/linux-libc-headers/4.1-r0/temp/log.do_fetch.60318
ERROR: Task 1430 (/home/sc/Apps/riscv-poky/build/../meta-riscv/recipes-kernel/linux-libc-headers/linux-libc-headers_4.1.bb, do_fetch) failed with exit code '1'
(Not sure about those ones too, I guess fixing warnings and previous error can solve this problem as well)

Any help will be appreciated,
Thanks

Sadullah

**Comment by scanakci** [comment]
Actually I noticed that I am getting three more warnings at the beginning:

WARNING: Failed to fetch URL http://zlib.net/pigz/pigz-2.3.3.tar.gz, attempting MIRRORS if available
WARNING: Failed to fetch URL http://download.gna.org/cryptodev-linux/cryptodev-linux-1.8.tar.gz, attempting MIRRORS if available
WARNING: Failed to fetch URL ftp://ftp.csx.cam.ac.uk/pub/software/programming/pcre/pcre-8.38.tar.bz2, attempting MIRRORS if available

pigz is updated to 2.3.4 as far as I see.
cryptodev-linux is 1.9 now
pcre is 8.40.

**Comment by scanakci** [comment]
After digging the issue for two days, I am able to figure out the problem. Here how I fixed the issue:
Firstly ignore all warnings, it is able to find required versions by looking mirrors.
Errors are originating due to riscv-linux repo. Commit revision,"174f39501397a32c3d9c3220e3b55ec20c16303f",  specified in the files given below does not belong to master branch:
riscv-poky/meta-riscv/recipes-kernel/linux-libc-headers/linux-libc-headers_4.1.bb
riscv-poky/meta-riscv/recipes-kernel/linux/ linux-riscv_4.1.bb 

I changed the branch name specified in SRC_URI to 'riscv-next' from 'master' in both files and successfully passed all the tests.
